model/storage-private/src/main/java/org/keycloak/storage/datastore/DefaultExportImportManager.java

```diff
@@ -860,11 +860,28 @@ public void updateRealm(RealmRepresentation rep, RealmModel realm) {
         session.clientPolicy().updateRealmModelFromRepresentation(realm, rep);
 
         if (rep.getSmtpServer() != null) {
-            Map<String, String> config = new HashMap(rep.getSmtpServer());
-            if (rep.getSmtpServer().containsKey("password") && ComponentRepresentation.SECRET_VALUE.equals(rep.getSmtpServer().get("password"))) {
-                String passwordValue = realm.getSmtpConfig() != null ? realm.getSmtpConfig().get("password") : null;
-                config.put("password", passwordValue);
+
+            Map<String, String> config = new HashMap<>(rep.getSmtpServer());
+
+            if(rep.getSmtpServer().containsKey("authType") && "basic".equals(rep.getSmtpServer().get("authType"))) {
+                if (rep.getSmtpServer().containsKey("password") && ComponentRepresentation.SECRET_VALUE.equals(rep.getSmtpServer().get("password"))) {
+                    String passwordValue = realm.getSmtpConfig() != null ? realm.getSmtpConfig().get("password") : null;
+                    config.put("password", passwordValue);
+                }
+                config.remove("authTokenUrl");
```

Comment:
Should the configuration be removed if another mode is selected? Would it not be more sensible to keep this information around and only use the `authType` as a switch?

Reply:
Imho the other values should be removed to be explicit and not keep unused stuff, but I can also follow the idea to keep theses.

Reply:
I am not sure if we do this anywhere else, I am not for or against it, but this kind of behavior should be consistent.